crates/rattler/src/install/mod.rs

```diff
@@ -248,6 +251,21 @@ pub async fn link_package(
     // Wrap the python info in an `Arc` so we can more easily share it with async tasks.
     let python_info = options.python_info.map(Arc::new);
 
+    let mut clobber_paths = HashSet::new();
+    // register all paths in the install driver path registry
+    {
+        let mut paths_registry = driver.locked_paths_registry();
+        for entry in paths_json.paths.iter() {
+            let path = entry.relative_path.clone();
+
+            if let Entry::Vacant(e) = paths_registry.entry(path.clone()) {
+                e.insert(index_json.name.as_normalized().to_string());
+            } else {
+                clobber_paths.insert(path);
+            }
+        }
+    }
```

Comment:
I think it makes sense to move this clobber registry into its own type and have a function on that type that does this. It splits up and compartmentalizes the logic a little better. 

I also feel like the overhead of cloning the package name hundreds of times (once for every path) is pretty resource heavy. If you create an object for the registry you could also have a lookup from index to name. You can then store indices in the map from path to package name instead.

Would also be nice if you could create the registry from an existing set of `PrefixRecords` to ensure you don't overwrite existing files maybe.

Reply:
yep, gonna take care of this next week!